enhancements/20190903-SRIOV-GA.md

```diff
@@ -0,0 +1,179 @@
+from user to enhance this document. This proposal is to provide DHCP
+configuration guide to help user understand DHCP CNI configuration and
+troubleshooting.
+
+##### Mac Address Pool Management
+
+Fixed MAC address can be specified via pod annotation for SR-IOV devices.
+However, user has to manually add MAC address for each SR-IOV interface
+in pod annotation which is error prone and inefficient. This proposal is
+to improve user experience by providing a MAC pool management tool which
+automatically assigns a MAC address in pod annotation.
+
+#### SR-IOV Operator
+
+Currently, the Custom Resources of SR-IOV Network Operator can only be
+validated by the declarative OpenAPI v3 schema validation rules defined
+in CRD and setting default value for CR is not supported. 
+
+To improve the user experience, an admission controller webhook for the
+CRD of SR-IOV network Operator shall be implemented. The admission controller
+webhook is expected to validate and set default value for the Custom Resource.
+
+The admission controller webhook and its RBAC shall be provisioned to
+openshift master nodes as a deployment by the SRIOV network operator.
+
+The admission controller webhook is expected to:
+- Set default value for fields of SriovNetwork CR, if not specified by user
+- Set default value for fields of SriovNetworkNodePolicy CR, if not specified
+  by user
+- Validate during SriovNetwork CR creation and updating request. The webhook
+  shall be able to validate following errors:
+  - `networkNamespace` refer to non-existing namespace
+- Validate during SriovNetworkNodePolicy CR creation and updating request
+  The webhook shall be able to validate following errors:
+  - The `numVFs` is larger than the maximum value that the NIC can support
+  - The `nicSelector` selects PFs which has already been covered by existing
+    Policy CRs
+  - The `resourceName` contains invalid characters
+  - `pfName` or `rootDevices` refers to non-existing device.
+
+
+#### SR-IOV QoS Flags
+
+Enable VF flags for security and QoS features.
+
+- Spoof Check
+- Trusted VF
+- Max rate limit
+- Min rate limit
+
+### Risks and Mitigations
+
+#### NUMA
+
+NUMA Awareness (aka Topology Manager) will not be available when SR-IOV GA.
+SR-IOV VF devices requested by pod will not be guaranteed to have NUMA
+locality preferences. Topology Manager will likely be included in 4.4 or 4.5
+which has a dependency on upstream development.
+
+## Design Details
+
+### Test Plan
+
+Build baremetal CI environment for the following tests:
+
+- SR-IOV Operator e2e test
+- SR-IOV CNI, Device Plugin and Admission Controller e2e test
+- Upgrade, negative and performance test
+
+### Graduation Criteria
+
+From Tech Preview to GA
+
+#### Tech Preview -> GA
+
+- Manage SR-IOV end to end deployment and configuration via Operator
+- Sufficient test coverage (upgrade, negative and performance)
+- IP and MAC address management for SR-IOV interfaces
+- Security and QoS support for SR-IOV interfaces
```

Comment:
Security and QoS flags are not a requirements for GA.

Reply:
Removed